tls/utils.py

```diff
@@ -24,3 +26,44 @@ def _decode(self, obj, context):
 
 def UBInt24(name):  # noqa
     return _UBInt24(construct.Bytes(name, 3))
+
+
+class BytesAdapter(construct.Adapter):
+
+    def _encode(self, obj, context):
+        if not isinstance(obj, bytes):
+            raise construct.AdaptationError("{} requires bytes, got {}".format(
+                self.subcon.name, repr(obj)))
+        return obj
+
+    def _decode(self, obj, context):
+        return obj
+
+
+def PrefixedBytes(name, length_field=construct.UBInt8("length")):  # noqa
+    return construct.LengthValueAdapter(
+        construct.Sequence(
+            name,
+            length_field,
+            BytesAdapter(
+                construct.Field("data",
+                                operator.attrgetter(length_field.name))))
+    )
+
+
+def TLSPrefixedArray(subconn, length_name="length"):  # noqa
```

Comment:
Same as above, can you please add a docstring? The description you wrote in the PR sounds like it could do well as a docstring. :)